flytekit/common/workflow.py

```diff
@@ -286,6 +287,20 @@ def register(self, project, domain, name, version):
             self._id = old_id
             raise
 
+    @_exception_scopes.system_entry_point
+    def serialize(self):
+        """
+        Serializing a workflow should produce an object similar to what the registration step produces, in preparation
+        for actual registration to Admin.
+
+        :rtype: flyteidl.admin.workflow_pb2.WorkflowSpec
```

Comment:
Also, if we're going to serialize, maybe go straight to bytes and call `SerializeToString()`?

Reply:
And if we want to keep it as the IDL object, let's refactor the engine to call this method so we don't have duplicate code.

Reply:
I'm gonna go with option 2 here.

Reply:
Actually, can I just leave it as is?  It's not quite the same (serialize calls to_flyteidl already).

So my plan is to remove the register call completely in a future PR (since registration happens from a container, we can control this, and only will do this after all refactorator PRs are done, or possibly rewrite how pyflyte register itself works).

But I also don't want bytes quite yet.  I haven't quite worked out how it'll work but I want the option to serialize these all to a pipe.  To do this, the serialize command will also need to emit the size of each object (since protobuf doesn't have a native delimiter).  I might try to do that then.  If i change it now then i can't use the current `write_proto_to_file` function too.

Reply:
you do you. i'm not sure if fully deprecating the register call is worth the pain though. Adding serialize is a nice option, but I don't think I see the value in removing the existing registration guts and then dealing with the migration.